lib/api/redfish-1.0/task-service.js

```diff
@@ -0,0 +1,145 @@
+    options.systems = [];
+
+    Promise.all([ dataFactory('', 'graphObjects') ])
+        .spread(function(graphs) {
+            _.forEach(graphs, function(graph) {
+                if (graph.systemId && -1 === options.systems.indexOf(graph.systemId))  {
+                    options.systems.push(graph.systemId);
+                }
+            });
+            options.graphs = graphs;
+            return redfish.render('redfish.1.0.0.taskservice.1.0.0.json', 
+                 'TaskService.1.0.0.json#/definitions/TaskService',
+                  options);
+    }).then(function(output) {
+        res.status(200).json(output);
+    })
+    .catch(function(error) {
+        res.status(500).json(error);
+    });
+}
+
+function listTasks(req, res) {
+    var options = {};
+    options.basepath = req.swagger.operation.api.basePath;
+    options.templateScope = ['global'];
+    options.url = req.url;
+    options.dateTime = moment.utc().format();
+
+    Promise.all([ dataFactory('', 'graphObjects') ])
+        .spread(function(graphs) {
+            options.graphs = graphs;
+            return redfish.render('redfish.1.0.0.taskcollection.json', 
+                 'TaskCollection.json#/definitions/TaskCollection',
+                  options);
+    }).then(function(output) {
+        res.status(200).json(output);
+    })
+    .catch(function(error) {
+        res.status(500).json(error);
+    });
+}
+
+function getTask(req, res) {
+    var identifier = req.swagger.params.identifier.value;
+    var options = {};
+    options.basepath = req.swagger.operation.api.basePath;
+    options.templateScope = ['global'];
+    options.url = req.url;
+    options.identifier = identifier;
+
+    Promise.all([ dataFactory('', 'graphObjects') ])
+        .spread(function(graphs) {
+            options.graphs = _.filter(graphs, function(graph) {
+                return graph.id === identifier;
+            });
+            options.graph = options.graphs[0];
+
+            return redfish.render('redfish.1.0.0.task.1.0.0.json', 
+                 'Task.1.0.0.json#/definitions/Task',
+                  options);
+    }).then(function(output) {
+        res.status(200).json(output);
+    })
+    .catch(function(error) {
+        res.status(500).json(error);
+    });
+}
+
+function getSystemTasks(req, res) {
+    var identifier = req.swagger.params.identifier.value;
+    var options = {};
+    options.basepath = req.swagger.operation.api.basePath;
+    options.templateScope = ['global'];
+    options.url = req.url;
+    options.identifier = identifier;
+
+    waterline.nodes.needByIdentifier(identifier)
+    .then(function() {
+        return Promise.all([ dataFactory('', 'graphObjects') ])
+            .spread(function(graphs) {
+                options.graphs = _.filter(graphs, function(graph) {
+                    return graph.systemId === identifier;
+                });
+                return redfish.render('redfish.1.0.0.taskcollection.json', 
```

Comment:
another dummy question - if the result is an empty set, does this end up returning a 404 response?